addons/crm_iap_lead/models/crm_iap_lead_helpers.py

```diff
@@ -0,0 +1,121 @@
+from math import floor, log10
+from odoo import api, models
+
+
+class CRMHelpers(models.Model):
+    _name = 'crm.iap.lead.helpers'
+    _description = 'Helper methods for crm_iap_lead modules'
+
+    @api.model
+    def notify_no_more_credit(self, service_name, model_name, notification_parameter):
+        """
+        Notify about the number of credit.
+        In order to avoid to spam people each hour, an ir.config_parameter is set
+        """
+        already_notified = self.env['ir.config_parameter'].sudo().get_param(notification_parameter, False)
+        if already_notified:
+            return
+        mail_template = self.env.ref('crm_iap_lead.lead_generation_no_credits')
+        iap_account = self.env['iap.account'].search([('service_name', '=', service_name)], limit=1)
+        # Get the email address of the creators of the records
+        res = self.env[model_name].search_read([], ['create_uid'])
+        uids = set(r['create_uid'][0] for r in res if r.get('create_uid'))
+        res = self.env['res.users'].search_read([('id', 'in', list(uids))], ['email'])
+        emails = set(r['email'] for r in res if r.get('email'))
+
+        mail_values = mail_template.generate_email(iap_account.id)
+        mail_values['email_to'] = ','.join(emails)
+        mail = self.env['mail.mail'].create(mail_values)
+        mail.send()
+        self.env['ir.config_parameter'].sudo().set_param(notification_parameter, True)
+
+    @api.model
+    def lead_vals_from_response(self, record, company_data, people_data):
```

Comment:
you cannot pass record as parameter, pass different values, lead_type, team_id, tag_ids and user_id as you are working on different models